app/src/main/java/com/example/habitsmasher/ui/history/HabitEventItemAdapter.java

```diff
@@ -0,0 +1,95 @@
+import com.firebase.ui.firestore.FirestoreRecyclerAdapter;
+import com.firebase.ui.firestore.FirestoreRecyclerOptions;
+import com.firebase.ui.firestore.ObservableSnapshotArray;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * The HabitEventItemAdapter class
+ * Based on HabitItemAdapter, extension of FirestoreRecycler that shows habit events list with live updates
+ */
+public class HabitEventItemAdapter extends FirestoreRecyclerAdapter<HabitEvent, HabitEventItemAdapter.HabitEventViewHolder> {
+    // Initialize variables
+    private Context _context;
+    private final String DATE_FORMAT = "dd/MM/yyyy";
+    private final ObservableSnapshotArray<HabitEvent> _snapshots;
+
+    /**
+     * Create a new RecyclerView adapter that listens to a Firestore Query.  See {@link
+     * FirestoreRecyclerOptions} for configuration options.
+     *
+     * @param options the firestore entities
+     */
+    public HabitEventItemAdapter(@NonNull FirestoreRecyclerOptions<HabitEvent> options) {
+        super(options);
+        _snapshots = options.getSnapshots();
+    }
+
+    @NonNull
+    @Override
+    public HabitEventViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        // Inflate view based on HabitEventViewHolder
+        _context = parent.getContext();
+        View view = LayoutInflater.from(_context).inflate(R.layout.habit_event_row, parent, false);
+        return new HabitEventViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull HabitEventViewHolder holder, int position, @NonNull HabitEvent habitEvent) {
+        // Set UI elements of habit event
+        holder._habitEventDate.setText(new SimpleDateFormat(DATE_FORMAT).format(habitEvent.getStartDate()));
+        holder._habitEventComment.setText(habitEvent.getComment());
+
+        // TODO: Implement image setting too
+    }
+
+    /**
+     * Necessary function for firestore recycler view
+     * @return number of elements in snapshots
+     */
+    @Override
+    public int getItemCount() { return _snapshots.size(); }
+
+    /**
+     * This class holds any necessary elements of the habit event on the front-end
+     */
+    public static class HabitEventViewHolder extends RecyclerView.ViewHolder {
+        // List of UI elements
+        private final TextView _habitEventDate;
+        private final TextView _habitEventComment;
+        // private final ImageView _habitEventImage;
+
+        /**
+         * Default constructor
+         * @param itemView (View): The habit event view
+         */
+        public HabitEventViewHolder(@NonNull View itemView) {
+            super(itemView);
+
+            _habitEventDate = itemView.findViewById(R.id.habit_event_date);
+            _habitEventComment = itemView.findViewById(R.id.habit_event_comment);
+            // _habitEventImage = itemView.findViewById(R.id.habit_event_image);
```

Comment:
unused code, make sure to delete if not needed for later

Reply:
Needed later :)